src/nodes.coffee

```diff
@@ -380,59 +381,63 @@ exports.Block = class Block extends Base
 
 #### Literal
 
-# Literals are static values that can be passed through directly into
-# JavaScript without translation, such as: strings, numbers,
+# `Literal` is a base class for static values that can be passed through
+# directly into JavaScript without translation, such as: strings, numbers,
 # `true`, `false`, `null`...
 exports.Literal = class Literal extends Base
   constructor: (@value) ->
 
-  makeReturn: ->
-    if @isStatement() then this else super
-
-  isAssignable: ->
-    IDENTIFIER.test @value
-
-  isStatement: ->
-    @value in ['break', 'continue', 'debugger']
-
   isComplex: NO
 
   assigns: (name) ->
     name is @value
 
+  compileNode: (o) ->
+    [@makeCode @value]
+
+  toString: ->
+    " #{if @isStatement() then super else @constructor.name}: #{@value}"
+
+exports.NumberLiteral = class NumberLiteral extends Literal
+
+exports.StringLiteral = class StringLiteral extends Literal
+
+exports.RegexLiteral = class RegexLiteral extends Literal
+
+exports.PassthroughLiteral = class PassthroughLiteral extends Literal
+
+exports.IdentifierLiteral = class IdentifierLiteral extends Literal
```

Comment:
No tests break if you change the next line to `isAssignable: YES`. Not sure what to do.